### PR TITLE
feat(monkeys): Interact with "external" users from the test [WPB-5129]

### PR DESCRIPTION
**Pull request by augustocdias**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

This adds 2 more actions to IM to accept and send connection request to users outside of the monkeys scope. Sending messages, adding to groups is already supported as long as monkeys and users are known to each other.

As discussed in #2172, unit tests are being removed in favour of e2e tests to be added later.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: monkeys/src/main/kotlin/com/wire/kalium/monkeys/actions/SendMessageAction.kt [hunk 1]
```diff
@@ -27,10 +27,87 @@ import com.wire.kalium.monkeys.pool.MonkeyPool
...
+       I'm in the mood for some bananas. Can you or your group help me out? It would mean a lot.
+
+       Thanks in advance! 
+    """,
+    """
+       Hey,
+
+       I've got this banana hankering that won't quit. Can you be the banana hero I need?
+
+       Much appreciated!
```
Comment (vitorhugods) [inline monkeys/src/main/kotlin/com/wire/kalium/monkeys/actions/SendMessageAction.kt:109]: LOL 🤣

File: monkeys/src/main/kotlin/com/wire/kalium/monkeys/actions/HandleExternalRequestAction.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+        Respect,
+        A neutral monkey 🍌
+    """.trimIndent(), """
+        Listen up,
+
+        I ain't messin' around. I want them bananas, and I want 'em now. You better deliver or there'll be consequences.
+
+        No games,
+        An evil monkey 🍌👿💀
```
Comment (vitorhugods) [inline monkeys/src/main/kotlin/com/wire/kalium/monkeys/actions/HandleExternalRequestAction.kt:47]: This is too good.

File: monkeys/src/main/kotlin/com/wire/kalium/monkeys/homeDirectory.kt [hunk 3]
```diff
@@ -29,13 +29,12 @@ fun coreLogic(
...
     val coreLogic = CoreLogic(
-        rootPath,
-        kaliumConfigs = KaliumConfigs(
+        rootPath, kaliumConfigs = KaliumConfigs(
             developmentApiEnabled = true,
             encryptProteusStorage = true,
             isMLSSupportEnabled = true,
             wipeOnDeviceRemoval = true,
-        ), "Wire Infinite Monkeys"
+        ), userAgent = "Wire Infinite Monkeys", useInMemoryStorage = false
```
Comment (vitorhugods) [inline monkeys/src/main/kotlin/com/wire/kalium/monkeys/homeDirectory.kt:37]: I think you can use `inMemoryStorage = true` now. After 

- #2274

**Review by vitorhugods — APPROVED** [review]
LGTM.

I just think that `inMemoryStorage` can be used again now that it supports login and logout.

File: monkeys/src/main/kotlin/com/wire/kalium/monkeys/homeDirectory.kt (hunk 3 above)
Comment (augustocdias) [reply monkeys/src/main/kotlin/com/wire/kalium/monkeys/homeDirectory.kt:37]: I'll enable it again :) Forgot to push

File: monkeys/src/main/kotlin/com/wire/kalium/monkeys/actions/HandleExternalRequestAction.kt (hunk 2 above)
Comment (augustocdias) [reply monkeys/src/main/kotlin/com/wire/kalium/monkeys/actions/HandleExternalRequestAction.kt:47]: All AI generated 🤣

File: monkeys/src/main/kotlin/com/wire/kalium/monkeys/pool/MonkeyPool.kt [hunk 4]
```diff
@@ -41,29 +50,36 @@ class MonkeyPool(users: List<UserData>, testCase: String) {
...
 
+    suspend fun warmUp(core: CoreLogic) = coroutineScope {
+        // this is needed to create key packages for clients at least once
+        poolById.values.map {
+            async {
+                it.login(core) {}
+                it.logout {}
+            }
+        }.awaitAll()
+    }
```
Comment (typfel) [inline monkeys/src/main/kotlin/com/wire/kalium/monkeys/pool/MonkeyPool.kt:81]: Do you have ticket for this bug? Let's add reference to it here

**Review by typfel — APPROVED** [review]

**Pull request merged**